app/services/IndividualDetailsService.scala

```diff
@@ -64,13 +63,22 @@ class IndividualDetailsServiceImpl @Inject()(
   override def getNPSPostCode(idData: IndividualDetails): String =
     getAddressTypeResidential(idData.addressList).addressPostcode.map(_.value).getOrElse("")
 
-  override def getIdData(pdvData: PDVResponseData)(implicit hc: HeaderCarrier): Future[Either[IndividualDetailsError, IndividualDetails]] = {
-    getIndividualDetails(IndividualDetailsNino(pdvData.personalDetails match {
-      case Some(data) => data.nino.nino
-      case None =>
-        logger.warn("No Personal Details found in PDV data, likely validation failed")
-        EmptyString
-    })).value
+  override def getIdData(pdvResponse: PDVResponse)(implicit hc: HeaderCarrier): Future[Either[IndividualDetailsError, IndividualDetails]] = {
+    pdvResponse match {
+      case PDVSuccessResponse(pdvData : PDVResponseData) =>
+        getIndividualDetails(IndividualDetailsNino(pdvData.personalDetails match {
+          case Some(data) => data.nino.nino
+          case None =>
+            logger.warn("No Personal Details found in PDV data, likely validation failed")
+            EmptyString
+        })).value
+      case _ =>
```

Comment:
I think we should check specifically for the PDV not found response here?

Reply:
other than PDV success, all fall into `case _ =>` scenario.